assignment/impl/src/java/org/sakaiproject/assignment/impl/persistence/AssignmentRepositoryImpl.java

```diff
@@ -143,12 +143,23 @@ public AssignmentSubmission findSubmission(String submissionId) {
         return geCurrentSession().get(AssignmentSubmission.class, submissionId);
     }
 
+    @Override
+    @Transactional
+    public AssignmentSubmission getSubmissionFromDatabase(String submissionId) {
+	Session session = geCurrentSession();
+        AssignmentSubmission submission = (AssignmentSubmission) session.get(AssignmentSubmission.class, submissionId);
+        session.refresh(submission);
+        return submission;
+    }
+
     @Override
     @Transactional
     public void updateSubmission(AssignmentSubmission submission) {
         if (existsSubmission(submission.getId())) {
             submission.setDateModified(Instant.now());
-            geCurrentSession().merge(submission);
+            Session session = geCurrentSession();
+            session.merge(submission);
```

Comment:
submission = session.merge(submission);

You also need to apply this patch in your instance: https://github.com/sakaiproject/sakai/commit/78a5732b4387ae4c202038bbe8e98627d9bd9460

Reply:
@jesusmmp - Thanks for reviewing this. 

The return type for the updateSubmission method I introduced has a return type of void. Are you implying that I should change this to AssignmentSubmission? I'm happy to do so, but just wanted to clarify your meaning.

Also, I already have the patch you cite within the SAK-48551 branch on which this PR is based. Let me know if I'm missing your meaning. Thanks.